UnityProject/Assets/Scripts/Objects/Doors/Modules/AccessModule.cs

```diff
@@ -59,30 +70,44 @@ public override ModuleSignal BumpingInteraction(GameObject byPlayer, HashSet<Doo
 			return ModuleSignal.Continue;
 		}
 
-		public override bool CanDoorStateChange()
-		{
-			return true;
-		}
 
 		private bool CheckAccess(GameObject player)
 		{
-			if (accessRestrictions.CheckAccess(player))
+			Inplayer = player;
+			master.HackingProcessBase.ImpulsePort(ProcessCheckAccess);
+
+			if (Inplayer == null)
 			{
+				Inplayer = null;
```

Comment:
Also you set Inplayer and then null it after ImpulsePort(), is that right? just checking

Reply:
Ah nvm i followed ImpulsePort()